### PR TITLE
Added instructions for handling release branches

### PR DESCRIPTION
**Pull request by mikejolley**
Adds some instructions to the contributing doc documenting how release branches are setup, and how to handle cherry picks.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,14 +50,41 @@ These instructions cover new releases of the blocks plugin for those with commit
...
+You'll also need to manually update the changelog and version numbers in the release branch before tagging. There is a changelog helper script which can list changelog entries from merged PRs in a milestone:
+
+`npm run changelog`
+
+Follow the instructions, then add the changelog entries to the readme.txt file before committing to the release branch. As well as the changelog, there are version strings which need updating in:
+
+-   `package.json`
+-   `readme.text`
+-   `src/package.php`
+-   `woocommerce-gutenberg-products-block.php`
```
Comment (nerrad) [inline CONTRIBUTING.md]: Version string updates should automatically get taken care of by the `version-changes.sh` script as part of the deploy script.

**Comment by nerrad** [comment]
Also, should we include some instructions about:

- merging readme.txt changes back to master
- for minor/major releases, manually bumping versions in master to the next minor release (+`-dev`)?

**Comment by mikejolley** [comment]
@nerrad That was partially covered already but added some more text just now.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,20 +70,15 @@ To do this:
...
-Follow the instructions, then add the changelog entries to the readme.txt file before committing to the release branch. As well as the changelog, there are version strings which need updating in:
+Follow the instructions, then add the changelog entries to the readme.txt file before committing to the release branch. Once all of this is done, you are safe to run the deploy scripts explained below.
 
--   `package.json`
--   `readme.text`
--   `src/package.php`
--   `woocommerce-gutenberg-products-block.php`
-
-Once all of this is done, you are safe to run the deploy scripts exaplained below.
+If you're using the GitHub deploy script you no longer need to worry about version bumps; it's automated. The `$ npm run deploy` (explained below) will update version strings in `package.json`, `readme.text`, `src/package.php`, and `woocommerce-gutenberg-products-block.php` for you, committing the changes to the current branch.
```
Comment (nerrad) [inline CONTRIBUTING.md]: Nit: should this be `src/Package.php`?

**Pull request merged**